src/opm/input/eclipse/Schedule/Well/Well.cpp

```diff
@@ -1426,6 +1426,10 @@ bool Well::handleWELSEGS(const DeckKeyword& keyword) {
         this->updateSegments(std::move(new_segments));
     } else
         this->updateSegments( std::make_shared<WellSegments>(keyword) );
+
+    // Ensure reference depth is depth of top segment node (as it should be)
+    this->ref_depth = this->segments->depthTopSegment();    
```

Comment:
Hi, @vkip, this should not do any harm. But at the same time, in the function 

```c++
void Well::updateSegments(std::shared_ptr<WellSegments> segments_arg) {
    this->segments = std::move(segments_arg);
    this->updateRefDepth( this->segments->depthTopSegment() );
}
```
the function `updateRefDepth( this->segments->depthTopSegment() )` is already called. 

```c++
bool Well::updateRefDepth(const std::optional<double>& ref_depth_arg) {
    if (this->ref_depth != ref_depth_arg) {
        this->ref_depth = ref_depth_arg;
        return true;
    }

    return false;
}
```

Do you see any situation the added line is necessary? Thanks.

Reply:
No.. somehow I missed this - sorry! I'll just close this one.